src/Rules/Keywords/DeclareStrictTypesRule.php

```diff
@@ -0,0 +1,69 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Keywords;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt;
+use PHPStan\Analyser\Scope;
+use PHPStan\Parser\DeclarePositionVisitor;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\UnionType;
+use function sprintf;
+
+/**
+ * @implements Rule<Stmt\Declare_>
+ */
+class DeclareStrictTypesRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return Stmt\Declare_::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$declaresStrictTypes = false;
+		foreach ($node->declares as $declare) {
+			if (
+				$declare->key->name !== 'strict_types'
+			) {
+				continue;
+			}
+
+			$validValues = new UnionType([new ConstantIntegerType(0), new ConstantIntegerType(1)]);
```

Comment:
Please don't use the type system for this, just the AST. I'm pretty sure some stuff is not allowed that would slip through this.